var/spack/repos/builtin/packages/rocsolver/package.py

```diff
@@ -48,7 +50,7 @@ class Rocsolver(CMakePackage):
                 '4.2.0', '4.3.0', '4.3.1']:
         depends_on('hip@' + ver, type='build', when='@' + ver)
         depends_on('comgr@' + ver, type='build', when='@' + ver)
```

Comment:
And this is likely not a direct dependency, but rather a transient dependency through hip, can you remove it if that's the case?

Reply:
Interesting. I've removed it. That also applies to numactl, which I have also removed. rocsolver for ROCm 4.3.1 builds and runs fine without them stated here. Should hsa-rocr-dev be removed as well, for the same reasons?

Reply:
Yeah, numactl is only there because *if* you build hip (or a dep of it i don't recall) statically, you have to link against numactl yourself when linking against the static hip libs. hipcc adds this -lnuma line then. So, it's just another transitive dependency, and I don't think we support static compilation of hip libs.

Of course, optimally we could leave numactl as a transitive link dep, and have cmake or pkgconfig or whatever add the -lnuma flag, not hipcc.

Reply:
However, in old versions, I think there was a bug where they added -lnuma unconditionally in hipcc... so that has to be checked.

Reply:
> Can you bring back numactl for 3.7.0 only, see: https://github.com/ROCm-Developer-Tools/HIP/blob/rocm-3.7.0/bin/hipcc#L842.

Sure. Though, I can't test this because [hsa-rocr-dev-3.7.0 fails to build](https://gist.github.com/cgmb/8ced6ba6ea7b952be3a21b4e3102affb).

Reply:
@srekolam? :D

Reply:
I wonder if we should continue to support it anyways, but that's up to AMD. Considering the fact that there has been a breaking change in the driver and we can't run old rocm anways, we may as well drop it?

Reply:
i think we can leave the numactl for 3.7.0 if that's only version for which it is applicable. i will see if i can fix the hsa-rocr-dev package error if possible some time later. i do know 3.7.0 is older but just thinking we can try to keep it working.

Reply:
i looked at rocsparse ,rocfft, i think we should be okay here.

Reply:
Just to add to the pile, I don't think any version of rocblas 4.0.0 or earlier will build, either. The `gfxNNN:xnack-` syntax was introduced in 4.1.0, but spack seems to be adding it in all versions. I'm seeing errors like this:
```
Tensile/lib/CMakeFiles/TensileHost.dir/build.make:148: recipe for target 'Tensile/lib/CMakeFiles/TensileHost.dir/source/Debug.cpp.o' failed
make[2]: *** [Tensile/lib/CMakeFiles/TensileHost.dir/source/Debug.cpp.o] Error 1
clang-12: error: Invalid offload arch combinations: gfx906 and gfx906:xnack- (For a specific processor, a feature should either exist in all offload archs, or not exist in any offload archs)
```